managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2060,14 +2060,14 @@ private void internalReadFromLedger(ReadHandle ledger, OpReadEntry opReadEntry)
             long lastValidEntry = -1L;
             long entryId = firstEntry;
             for (; entryId <= lastEntry; entryId++) {
-                if (!opReadEntry.skipCondition.test(PositionImpl.get(ledger.getId(), entryId))) {
-                    if (firstValidEntry == -1L) {
-                        firstValidEntry = entryId;
-                    }
-                } else {
+                if (opReadEntry.skipCondition.test(PositionImpl.get(ledger.getId(), entryId))) {
                     if (firstValidEntry != -1L) {
                         break;
                     }
+                } else {
+                    if (firstValidEntry == -1L) {
+                        firstValidEntry = entryId;
+                    }
                 }
 
                 if (firstValidEntry != -1L) {
```

Comment:
It looks like this logic just works for the `else` branch. Can we merge it?

Reply:
Yes, we can move it to the `else` branch